File: src/DClare.Sdk/Models/Authentication/OAuth2AuthenticationSchemeDefinition.cs
namespace DClare.Sdk.Models.Authentication;

/// <summary>
/// Represents the definition of an OAuth 2.0 authentication scheme, including required endpoints and flow configuration.
/// </summary>
[Description("Represents the definition of an OAuth 2.0 authentication scheme, including required endpoints and flow configuration.")]
[DataContract]
public record OAuth2AuthenticationSchemeDefinition
    : OAuth2AuthenticationSchemeDefinitionBase
{

    /// <inheritdoc/>
    [IgnoreDataMember, JsonIgnore, YamlIgnore]
    public override string Scheme => AuthenticationScheme.OAuth2;

    /// <summary>
    /// Gets or sets the configuration of the OAuth 2.0 endpoints to use for authorization, token exchange, and optionally revocation or introspection.
    /// </summary>
    [Description("Configuration of the OAuth 2.0 endpoints, including authorization and token URLs.")]
    [DataMember(Name = "endpoints", Order = 2), JsonPropertyName("endpoints"), JsonPropertyOrder(2), YamlMember(Alias = "endpoints", Order = 2)]
    public virtual OAuth2AuthenticationEndpointsDefinition Endpoints { get; set; } = new();

}

File: src/DClare.Sdk.TypeScript/models/o-auth2-authentication-scheme-definition.ts
import { OAuth2AuthenticationSchemeDefinitionBase } from './o-auth2-authentication-scheme-definition-base';import { OAuth2AuthenticationEndpointsDefinition } from './o-auth2-authentication-endpoints-definition';

/**
 * Represents the definition of an OAUTH2 authentication scheme
 */
export class OAuth2AuthenticationSchemeDefinition extends OAuth2AuthenticationSchemeDefinitionBase
 {
  /**
   */
  scheme: string;
  /**
   * Gets/sets the configuration of the OAUTH2 endpoints to use
   */
  endpoints: OAuth2AuthenticationEndpointsDefinition;
}